GiftModels/PremiumDetails.cs

```diff
@@ -41,8 +37,9 @@ public string PremiumChart
             {
                 if (string.IsNullOrWhiteSpace(_premiumChart) && string.IsNullOrWhiteSpace(Xcomment))
                     return null;
-                if (string.IsNullOrWhiteSpace(_premiumChart) && Xcomment.Contains("-"))
-                    return Xcomment.Substring(0, Xcomment.IndexOf('-'));
+                if (string.IsNullOrWhiteSpace(_premiumChart))
+                    InitAccountDetails();

```

Comment:
@taylorkj For clarity, I prefer to always wrap if statements with {} even if it is a single line. But I don't know if that is one of our preferences for our group.